crates/api_common/src/utils.rs

```diff
@@ -566,6 +567,28 @@ pub fn check_private_instance_and_federation_enabled(
   }
 }
 
+/// Read the site for an actor_id, but don't include it if its a local actor.
+///
+/// Used for GetCommunityResponse and GetPersonDetails
+pub async fn read_site_for_actor(
+  actor_id: DbUrl,
+  context: &LemmyContext,
+) -> Result<Option<Site>, LemmyError> {
+  let site_id = Site::instance_actor_id_from_url(actor_id.clone().into());
+  let site = Site::read_from_apub_id(&mut context.pool(), &site_id.into()).await?;
+  // no need to include metadata for local site (its already available through other endpoints).
+  // this also prevents us from leaking the federation private key.
+  if let Some(s) = &site {
+    if s.actor_id.domain() == Some(context.settings().hostname.as_ref()) {
+      Ok(None)
```

Comment:
You can simply check `community.local` or `person.local` (easiest inside read_community() and read_person() handlers). If its local then skip the read_site call.